pywbem/_logging.py

```diff
@@ -0,0 +1,407 @@
+        else:
+            if log_dest not in LOG_DESTINATIONS:
+                raise ValueError('Invalid log destination %s. valid log '
+                                 'destinations are: %s' %
+                                 (log_dest, LOG_DESTINATIONS))
+            if log_component not in LOG_COMPONENTS:
+                raise ValueError('Invalid log component %s. Valid log '
+                                 'components are: %s' %
+                                 (log_component, LOG_COMPONENTS))
+            if not log_level:
+                log_level = DEFAULT_LOG_LEVEL
+            if log_level not in LOG_LEVELS:
+                raise ValueError('Invalid log level %s. Valid log levels are:'
+                                 ' %s' %
+                                 (log_level, LOG_LEVELS))
+
+            if not log_detail_level:
+                log_detail_level = DEFAULT_LOG_DETAIL_LEVEL
+            if log_detail_level not in LOG_DETAIL_LEVELS:
+                raise ValueError('Invalid log detail %s. Valid Log detail '
+                                 'levels are %s' %
+                                 (log_detail_level, LOG_DETAIL_LEVELS))
+
+            if log_dest == 'stderr':
+                handler = logging.StreamHandler()
+                format_string = '%(asctime)s-%(name)s-%(message)s'
+            elif log_dest == 'file':
+                if not log_filename:
+                    raise ValueError('Filename required if log destination '
+                                     'is "file"')
+                handler = logging.FileHandler(log_filename)
+                format_string = '%(asctime)s-%(name)s-%(message)s'
+            else:
+                assert(log_dest == 'none')
+                handler = None
+                format_string = None
+
+            # set the logger name based on the log_component.
+            if log_component == 'http':
+                logger_name = LOG_HTTP_NAME
+                if log_level is None:
+                    log_level = 'debug'
+            elif log_component == 'ops':
+                logger_name = LOG_OPS_CALLS_NAME
+                if log_level is None:
+                    log_level = 'debug'
+            else:
+                raise ValueError('Invalid log_component %s' % log_component)
+
+            # Set the log level based on the log_level input
+            level = getattr(logging, log_level.upper(), None)
+            # check logging log_level_choices have not changed from expected
+            assert isinstance(level, int)
+            if level is None:
+                raise ValueError('Invalid log level %s specified. Must be one '
+                                 'of %s.' % (log_level, LOG_LEVELS))
+
+            # create named logger
+            if handler:
+                handler.setFormatter(logging.Formatter(format_string))
+                logger = logging.getLogger(logger_name)
+                logger.addHandler(handler)
```

Comment:
Using `addHandler() ` adds a second handler if this method is invoked a second time for the same log_component. This should be replaced with whatever is needed to replace the existing handler.